mlrose/opt_probs.py

```diff
@@ -275,13 +275,69 @@ def __init__(self, length, fitness_fn, maximize=True, max_val=2):
     def eval_node_probs(self):
         """Update probability density estimates.
         """
-        # Create mutual info matrix
-        mutual_info = np.zeros([self.length, self.length])
-        for i in range(self.length - 1):
-            for j in range(i + 1, self.length):
-                mutual_info[i, j] = -1 * mutual_info_score(
-                    self.keep_sample[:, i],
-                    self.keep_sample[:, j])
+        if (self.mimic_speed == False):
+            # Create mutual info matrix
+            mutual_info = np.zeros([self.length, self.length])
+            for i in range(self.length - 1):
+                for j in range(i + 1, self.length):
+                    mutual_info[i, j] = -1 * mutual_info_score(
+                        self.keep_sample[:, i],
+                        self.keep_sample[:, j])
+
+        elif (self.mimic_speed == True):
+            # Set ignore error to ignore dividing by zero
+            np.seterr(divide='ignore', invalid='ignore')
+
+            # get length of the sample which survived from mimic iteration
+            len_sample_kept = self.keep_sample.shape[0]
+            # get the length of the bit sequence / problem size
+            len_prob = self.keep_sample.shape[1]
+
+            # Expand the matrices to so each row corresponds to a row by row combination of the list of samples
+            b = np.repeat(self.keep_sample, self.length).reshape(len_sample_kept, len_prob * len_prob)
+            d = np.hstack(([self.keep_sample] * len_prob))
+
+            # Compute the mutual information matrix in bulk, by iterating through the list of possible feature values ((max_val-1)^2).
```

Comment:
Ideally, I would like the code for this package to be PEP8 compliant, which means that all lines of code need to be < 79 characters long. I think there are a few lines here that exceed that limit.

Reply:
Updated length, variable names, as well as line spacing for PEP8 compliance